keras_cv/layers/preprocessing/fmix.py

```diff
@@ -0,0 +1,207 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class FMix(tf.keras.__internal__.layers.BaseImageAugmentationLayer):
+    """FMix implements the FMix data augmentation technique.
+
+    Args:
+        alpha: Float value for beta distribution.  Inverse scale parameter for the gamma
+            distribution.  This controls the shape of the distribution from which the
+            smoothing values are sampled.  Defaults 0.5, which is a recommended value
+            in the paper.
+        decay_power: A float value representing the decay power for frequency decay prop
+            1/f**d.  Defaults to 3, as recommended in the paper.
+        max_soft: Float value representing softening value between 0 and 0.5 which
+            smooths hard edges in the mask.  Defaults to 0.0.
+        seed: Integer. Used to create a random seed.
+    References:
+        [FMix paper](https://arxiv.org/abs/2002.12047).
+
+    Sample usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    fmix = keras_cv.layers.preprocessing.mix_up.FMix(0.5)
+    augmented_images, updated_labels = fmix({'images': images, 'labels': labels})
+    # output == {'images': updated_images, 'labels': updated_labels}
+    ```
+    """
+
+    def __init__(self, alpha=0.2, decay_power=3, max_soft=0.0, seed=None, **kwargs):
+        super().__init__(seed=seed, **kwargs)
+        self.alpha = alpha
+        self.decay_power = decay_power
+        self.max_soft = max_soft
+        self.seed = seed
+
+    @staticmethod
+    def _sample_from_beta(alpha, beta, shape):
+        sample_alpha = tf.random.gamma(shape, 1.0, beta=alpha)
+        sample_beta = tf.random.gamma(shape, 1.0, beta=beta)
+        return sample_alpha / (sample_alpha + sample_beta)
+
+    def fftfreq(self, n, d=1):
+        # This function returns the sample frequencies of a discrete fourier transform.
+
+        N = tf.convert_to_tensor((n - 1) / 2 + 1)
```

Comment:
Rename `N` to something else to avoid upper casing.  Also will make code less confusing.  Basically it looks like `N` is sampling the number of frequencies required to capture all information in an FFT (because the result of the FFT is mirrored).

Do we need the convert to tensor call?

Reply:
Added this as part of the range itself.